### ISSUE TITLE
Recalculate comments doesn't work

### ISSUE THREAD

**Issue by Jimmi08** [e107inc/e107#4568]
## Bug Description
News displays a wrong number of comments.  
It was 15. I added a new one, it changed to 16. So it is not hardcoded.

comment manager displays only one record for news

Used recalculation, no change.

## How to Reproduce
Set manually news_comment_total in database to any value. 
Try to fix the number of comments

## Screenshots
![image](https://user-images.githubusercontent.com/5429548/133472239-94b3ce98-6758-4fae-8561-c225f2c43720.png)

![image](https://user-images.githubusercontent.com/5429548/133472390-df1ed610-a91e-4840-8e90-ef26f80fb364.png)

![image](https://user-images.githubusercontent.com/5429548/133472477-b93d242e-ca04-4354-949f-295f6000d837.png)

 

### PHP Version

```
 
PHP Version 7.4
```

**Comment by Deltik** [comment]
The comment count recalculator that is currently implemented [only updates the values in `e107_user.user_comments`](https://github.com/e107inc/e107/pull/3250/files#diff-3005e0dbe7842ef025bf82ab3c7e505e07e5b6d9d9dc8b1af48fd5bd5b7e0b03R213-R227).

I'm not aware of any feature that can fix the values in `e107_news.news_comment_total`.

I do consider this issue a valid bug, though, because I would expect that "Recalculate comment count" updates `e107_news.news_comment_total`, too.

**Comment by Jimmi08** [comment]
Ugh,  I wouldn't suppose something like that at all. Thanks for this note. 

This field is there because of speed?  How should other types work (custom plugin table using comment system) Or how to implement recalculation for custom types? Normally I would expect a solution in the e_comment addon, but it is not needed/used anymore.  
Maybe it is not needed anymore and that shortcode should return the actual count of comments (without news table field, then recalculation is not needed). 
Or at least change the message and add there ... User comments count...  

Another solution is to have the number of comments editable in news -  I have no idea why this happened but this way it would be quick fix.  For me, this would be enough.

**Comment by Deltik** [comment]
In my opinion, the cache at `e107_news.news_comment_total` is useless mostly because [`comment::getComments()` counts the comments anyway](https://github.com/e107inc/e107/blob/6fa4982d30362729c7a87fbb303cefedad25f4fc/e107_handlers/comment_class.php#L1228-L1248)!

**Comment by Jimmi08** [comment]
I agree. At least when a new comment is added, it could set this value, not always rise the existing value.
